src/FlipMove.js

```diff
@@ -599,6 +608,25 @@ class FlipMove extends Component<void, ConvertedProps, FlipMoveState> {
     return this.state.children.find(child => getKey(child) === key);
   }
 
+  hasChildData(key: string): boolean {
+    return Object.prototype.hasOwnProperty.call(this.childrenData, key);
+  }
+
+  getChildData(key: string): NodeData {
+    return this.childrenData[key] || {};
+  }
+
+  putChildData(key: string, data: NodeData): void {
+    this.childrenData[key] = {
+      ...this.childrenData[key] || {},
```

Comment:
I think this needs to be changed to `...(this.childrenData[key] || {})`.

The reason is that the spread operator is much lower precedence than the `||` operator ([source](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Operator_Precedence))

Actually, I think this could be improved as:

```js
putChildData(key: string, data: NodeData): void {
  this.childrenData[key] = {
    ...this.getChildData(key),
    data,
  }
}
```